### ISSUE TITLE
Add specific repository for tracking machine configurations

### ISSUE THREAD

**Issue by soroshsabz** [microsoft/devhome#1813]
### Suggested new feature or improvement

ITNOA

I think it is good to exist specific repository for tracking machine configuration

### Scenario

In machine configuration section we able to set specific repository for tracking machine configuration, and all of changes in machine configuration, commits to this repository automatically and if this repository connect to GitHub or AzureDevOps sync with upstream automatically after some confirmation by user

**Comment by soroshsabz** [comment]
related to #1812

**Comment by soroshsabz** [comment]
@cinnamon-msft No, we suggest in this issue, Dev Home use specific repository to store all configuration about this machine, and after any changes on this machine (for example install new program, or add new repository, or change Dev Drive properties and etc.) automatically commits these changes on this repository

as you know, Dev Home can apply some configuration on this machine with yaml files, so I suggest a systematic way to store and organize these yaml files

this functionally can provide very developer way to sync machine state (such as infrastructure as code) and move between states with git functionality easily

**Comment by denelon** [comment]
We're looking into building a tracking database via WinGet to record when configurations are applied, but users may still perform most operations outside of this mechanism, so it wouldn't necessarily be a reliable mechanism to detect the current state of the device. 

For that we're tracking:
* microsoft/winget-cli/issues/3274
